src/plugins/discover/public/application/angular/discover.js

```diff
@@ -633,12 +633,18 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
 
   const init = _.once(() => {
     $scope.updateDataSource().then(async () => {
-      const searchBarChanges = merge(data.query.state$, $fetchObservable).pipe(debounceTime(100));
+      const fetch$ = merge(
```

Comment:
Instead of reverting to using individual observables and relying on the `handleRefresh` (which can actually be removed) I'm suggesting the fix in the attached patch file.

[80635.txt](https://github.com/elastic/kibana/files/5390909/80635.txt)

Reply:
Thanks for the patch, but I think it would cause issue with excessive refetch on removing disabled filter which I mentioned in the description: 

> Also implicitly fixes a subtle bug with excessive re-fetch after deleting already disabled filter

Also some thoughts on query.state$ vs fetch$: https://github.com/elastic/kibana/issues/80586#issuecomment-709022740

Reply:
Agreed. Thanks for the feedback.